setup.py

```diff
@@ -1033,17 +1050,29 @@ def replace_extra_requirement_with_provider_packages(extra: str, providers: List
             ['simple-salesforce>=1.0.0', 'tableauserverclient']
 
     So transitively 'salesforce' extra has all the requirements it needs and in case the provider
-    changes it's dependencies, they will transitively change as well.
+    changes its dependencies, they will transitively change as well.
 
     In the constraint mechanism we save both - provider versions and it's dependencies
     version, which means that installation using constraints is repeatable.
 
+    For K8s, Celery and Dask which are both "Core executors" and "Providers" we have to
+    add the base dependencies to the core as well - in order to mitigate problems where
+    newer version of provider will have less strict limits. This should be done for both:
+    extras and their deprecated aliases. This is not a full protection however, the way
+    extras work, this will not add "hard" limits for airflow and the user who does not use
+    constraints
+
     :param extra: Name of the extra to add providers to
     :param providers: list of provider ids
     """
-    EXTRAS_REQUIREMENTS[extra] = [
-        get_provider_package_from_package_id(package_name) for package_name in providers
-    ]
+    if extra in ['cncf.kubernetes', 'kubernetes', 'celery']:
```

Comment:
We mention dask in the docstring, but don't actually do anything for it here? Which is right?

Reply:
Ah. Yeah. DASK is not a provider :)